src/common/components/layouts/pages/inspect/tree/Tree.js

```diff
@@ -60,11 +65,47 @@ export const errorMessagesMap = {
     },
 };
 
-function Tree({ ruleSummaries, expandedRules, onExpandRule, selectedCheck, setSelectedCheck, errorsMap, profile }) {
+function Tree({
+    ruleSummaries,
+    tags,
+    expandedRules,
+    onExpandRule,
+    selectedCheck,
+    setSelectedCheck,
+    errorsMap,
+    profile,
+}) {
     const [language, setLanguage] = useState(getItem(LS_ERROR_MESSAGES_LANGUAGE) || languageEnum.English);
     const [anchorMenuEl, setAnchorMenuEl] = useState(null);
+    const [isFilterOpen, setIsFilterOpen] = useState(false);
+    const [ruleSummariesFiltered, setRuleSummariesFiltered] = useState(ruleSummaries);
     const prevSelectedCheck = usePrevious(selectedCheck);
 
+    const isSubarray = useCallback((arr, subArr) => {
+        const hash = arr.reduce((acc, value) => {
+            acc[value] = true;
+            return acc;
+        }, {});
+        return subArr.every(el => el in hash);
+    }, []);
+    const onRuleFilter = useCallback(
+        ({ enabled, filteredTags }) => {
+            if (!filteredTags.length) {
+                setRuleSummariesFiltered(ruleSummaries);
+                return;
+            }
+            const ruleSummariesFiltered = ruleSummaries.map(rule => {
```

Comment:
The algorithm can be simplified using lodash function difference: `_.difference(filteredTags, rule.tags);`
If show all is selected, difference should be empty, otherwise difference length should be equal to filteredTags length. 

Custom `isSubarray` check is not needed in such case